File: Leetcode/Medium/productOfArrayExceptSelf.js
const nums = [1,2,3,4]

let res = []
let preNum = 1
for (let i=0; i < nums.length; i++){
    res[i] = preNum
    preNum = preNum * nums[i]
}

let postNum = 1
for (let i=nums.length -1; i>=0; i--){
    res[i] = res[i] * postNum
    postNum = postNum * nums[i]
}

console.log(res)

File: Leetcode/Medium/productOfArrayExceptSelf.py
nums = [1,2,3,4]
# nums = [-1,1,0,-3,3]
res = [1] * (len(nums))

preNum = 1
for i in range(len(nums)):
    res[i] = preNum
    preNum *= nums[i]

postNum = 1
for i in range(len(nums) -1, -1, -1):
    res[i] *= postNum
    postNum *= nums[i]

print(res)